earthpy/clip.py

```diff
@@ -34,6 +35,41 @@ def _clip_points(shp, clip_obj):
     return shp[shp.geometry.intersects(poly)]
 
 
+def _clip_multi_point(shp, clip_obj):
+    """Clip multi point features to the clip_obj GeoDataFrame extent.
+
+    Clip an input multi point to the polygon extent of the clip_obj
+    parameter. Points that intersect the clip_obj geometry are
+    extracted with associated attributes returned.
+
+    Parameters
+    ----------
+    shp : GeoDataFrame
+        multipoint geometry that is clipped to clip_obj.
+
+    clip_obj : GeoDataFrame
+        Reference polygon for clipping.
+
+    Returns
+    -------
+    GeoDataFrame
+        The returned GeoDataFrame is a clipped subset of shp
+        containing multi-point and point features.
+    """
+
+    if (
```

Comment:
This has some duplication (`shp["geometry"].iloc[0].type`) but you could do something like the following: 

```python
if "Point" in shp["geometry"].iloc[0].type
```

Reply:
ok.. the above didn't work but i could do it using a list comprehension. i'm slightly worried about how that will scale but we know that this doesn't scale as well as it could in parallel anyway and that people are working on that for geopandas so for now a list comprehension will do!

Reply:
That's weird - I implemented that change and all of the tests still pass, i.e., 

```python
def _clip_multi_point(shp, clip_obj):
    """..."""

    if "Point" in shp["geometry"].iloc[0].type:
        clipped = _clip_points(shp.explode().reset_index(level=[1]), clip_obj)
        clipped = clipped.dissolve(by=[clipped.index]).drop(columns="level_1")[
            shp.columns.tolist()
        ]
    return clipped
```

But, on second thought, would this function make more sense without the if statement? If the condition is not satisfied, then it will raise an exception when trying to return `clipped`, right?

Reply:
yes... well i realized two things that don't work
1. it just looks at the first geom_type but there could be many features so doing [0] skips that . thus a list comprehension is desired.
BUT
2. i just realized when the function is called, it is already assuming there is atleast one multipoint in the GDF SO... i'm thinking rather than the list comprehension should jsut drop the conditional altogether! 

```
    if any(shp.geometry.type == "MultiPoint"):
        return _clip_multi_point(shp, clip_obj)
```
so i thijnk it might make sense to just remove the check altogether for this particular helper function since it's being called on the condition that a multipoint feature exists already.

Reply:
Ok this has been cleaned up. i think the simpler version is better!! it won't hit this helper unless the conditional in the main function has been satisfy so there is no need to test it again as far as i can see.